code/_helpers/lists.dm

```diff
@@ -856,3 +856,21 @@ var/global/list/json_cache = list()
 /// Is this a dense (all keys have non-null values) associative list with at least one entry?
 /proc/is_dense_assoc(var/list/L)
 	return length(L) > 0 && !isnull(L[L[1]])
+
+// I have a feeling this is a redundant proc but I couldn't find another implementation.
+/proc/lists_are_equivalent(list/A, list/B, associative)
```

Comment:
/proc/same_entries` on line 209 of this file

Reply:
actually that doesn't properly handle the recursive case, so it should prboably be edited to do `(islist(first[entry]) && islist(second[entry])) ? same_entries(first[entry], second[entry]) : first[entry == second[entry]` or something